packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_CustomMeshFactory.cpp

```diff
@@ -326,50 +326,47 @@ namespace panzer_stk {
   void
   CustomMeshFactory::fillSolutionFieldData(STK_Interface &mesh) const
   {
+    constexpr std::size_t dim_1 = 8;
+
     for (int blk=0;blk<NumBlocks_;++blk) {
 
       std::stringstream block_id;
       block_id << "eblock-" << blk;
-      
+
       // elements in this processor for this block
-      std::vector<stk::mesh::Entity> elements;    
+      std::vector<stk::mesh::Entity> elements;
       mesh.getMyElements(block_id.str(), elements);
 
       // size of elements in the current block
-      std::size_t n_elements = elements.size();
-      
+      const auto n_elements = elements.size();
+
       // build local element index
       std::vector<std::size_t> local_ids;
-      for (std::vector<stk::mesh::Entity>::const_iterator
-             itr=elements.begin();itr!=elements.end();++itr) 
-        local_ids.push_back(mesh.elementLocalId(*itr));
-
-      // re-index solution fields in the same order of local_ids
-      std::vector<double> charge_density_by_local_ids, electric_potential_by_local_ids;
-      for (std::vector<stk::mesh::Entity>::const_iterator
-             itr=elements.begin();itr!=elements.end();++itr) {
-        int q = mesh.elementGlobalId(*itr) - OffsetToGlobalElementIDs_;
-        for (int k=0;k<8;++k) {
-          int loc = q*8 + k;
-          charge_density_by_local_ids.push_back(ChargeDensity_[loc]);
-          electric_potential_by_local_ids.push_back(ElectricPotential_[loc]);
+
+      Kokkos::View<double**, Kokkos::HostSpace> charge_density_by_local_ids("charge_density_by_local_ids", n_elements, dim_1);
+      Kokkos::View<double**, Kokkos::HostSpace> electric_potential_by_local_ids("electric_potential_by_local_ids", n_elements, dim_1);
+
```

Comment:
@rppawlo do you know if we need a layout here?  I don't think so but not 100%

Reply:
I don't think so. The special layouts are only needed for FAD scalar types.